features/swap/hooks/useTxRates.ts

```diff
@@ -66,23 +51,22 @@ export const useTxRates = ({
         )
   )
 
+  const [tokenBDollarValue] = useTokenDollarValue(tokenBSymbol)
   const conversionRateInDollar = usePersistance(
     isLoading || fetchingTokenDollarPrice || !shouldShowRates
       ? undefined
-      : protectAgainstNaN(
-          calculateTokenToTokenConversionDollarRate({
-            tokenAAmount,
-            conversionRate,
-            tokenADollarPrice,
-            oneTokenToTokenPrice,
-          })
-        )
+      : protectAgainstNaN(conversionRate * tokenBDollarValue)
   )
 
+  const tokenASwapValueUSD = (tokenADollarValue || 0) * (tokenAAmount || 0)
+  const tokenBSwapValueUSD =
+    (tokenBDollarValue || 0) * (tokenAAmount * conversionRate || 0)
```

Comment:
`tokenAAmount` here be `(tokenAAmount || 0)`?

Reply:
Hey `undefined` *  any number is NaN and (NaN || 0) is 0. `null` * any number is 0 so I think we good.